### PR TITLE
review feedback from jzila in #1561

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: libfs/fs.go [hunk 1]
```diff
@@ -855,16 +855,17 @@ func (o folderHandleChangeObserver) TlfHandleChange(
 	o()
 }
 
-// SubscribeToEndOfLife returns a channel that will be closed when this *FS
+// SubscribeToObsolete returns a channel that will be closed when this *FS
 // reaches end-of-life, meaning if user of this object caches it for long term
```
Comment (jzila) [inline libfs/fs.go]: s/end-of-life/obsolescence

File: libmime/patch.go [hunk 2]
```diff
@@ -18,6 +18,16 @@ func dontOverride(ext string, mimeType string) (newExt string, newMimeType strin
 // Provide a non-nil additional map (ext->mimeType) to add additional mime
 // types. Provide a non-nil Overrider to override any mime type defined in the
 // list. Note that the Overrider can override what's in the additional too.
+//
+// Overrider is different from additional in the way that, additional provides
+// exact ext-mimeType paris, while overrider allows the user of this function
```
Comment (jzila) [inline libmime/patch.go]: s/paris/pairs

File: libmime/patch.go [hunk 3]
```diff
@@ -18,6 +18,16 @@ func dontOverride(ext string, mimeType string) (newExt string, newMimeType strin
...
+// to examine ext-mimeTypes flexibly. For example, this allow overrider to
+// replace mimeTypes without an exausted list of all extensions that resolve to
```
Comment (jzila) [inline libmime/patch.go]: s/exhausted/exhaustive

File: libmime/patch.go [hunk 4]
```diff
@@ -18,6 +18,16 @@ func dontOverride(ext string, mimeType string) (newExt string, newMimeType strin
...
+// it. So why is additional useful? Go's mime package loads mime types from a
+// few filesystem locations such as /etc/apache2/mime.types . This happens
```
Comment (jzila) [inline libmime/patch.go]: `mime.types.`

File: libmime/patch.go [hunk 5]
```diff
@@ -18,6 +18,16 @@ func dontOverride(ext string, mimeType string) (newExt string, newMimeType strin
 // Provide a non-nil additional map (ext->mimeType) to add additional mime
 // types. Provide a non-nil Overrider to override any mime type defined in the
 // list. Note that the Overrider can override what's in the additional too.
+//
+// Overrider is different from additional in the way that, additional provides
+// exact ext-mimeType paris, while overrider allows the user of this function
+// to examine ext-mimeTypes flexibly. For example, this allow overrider to
```
Comment (strib) [inline libmime/patch.go]: "allow" -> "allows"

**Pull request merged**